apps/impulse/lib/screens/profile_screen.dart

```diff
@@ -276,8 +289,25 @@ class _ProfileScreenState extends State<ProfileScreen> {
                   padding: const EdgeInsets.all(paddingMedium),
                   child: Center(
                     child: ElevatedButton(
-                      onPressed: () {
-                        Navigator.pop(context);
+                      onPressed: () async{
+                        if (isCreating) {
+                          await ImpulseService().createUser(user).then((value) {
+                            userNotifier.setUser(user: user);
+                            Navigator.push(
+                              context,
+                              MaterialPageRoute(builder: (context) => const HomeScreen()),
+                            );
+                          });
+                        }
+                        else {
+                          await ImpulseService().updateUser(user).then((value) {
+                            userNotifier.setUser(user: user);
+                            Navigator.push(
+                              context,
+                              MaterialPageRoute(builder: (context) => const HomeScreen()),
+                            );
+                          });
+                        }
```

Comment:
On error there needs to be some feedback maybe in the snackbar, currently the push doesn't happen and the create should push and replace
```suggestion
                      onPressed: () async{
                        if (isCreating) {
                          await ImpulseService().createUser(user).then((value) {
                            userNotifier.setUser(user: user);
                            Navigator.pushReplacement(
                              context,
                              MaterialPageRoute(builder: (context) => const HomeScreen()),
                            );
                          });
                        }
                        else {
                          await ImpulseService().updateUser(user).then((value) {
                            userNotifier.setUser(user: user);
                            Navigator.push(
                              context,
                              MaterialPageRoute(builder: (context) => const HomeScreen()),
                            );
                          });
                        }
```